packages/salesforcedx-vscode-core/src/deploys/deployUtil.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+export const supportedToolingTypes = new Map([
+  ['Apexclass', 'ApexClassMember']
+]);
+
+/**
+ * Enum that represents the status of a Tooling Container Deploy
+ */
+export enum DeployStatusEnum {
+  Completed = 'Completed',
+  Queued = 'Queued',
+  Error = 'Error',
+  Failed = 'Failed'
+}
+
+// what other parameters would be necessary
+export interface FilePathOpts {
+  filepath: string;
+  wait?: string;
+}
+
+export interface ManifestOpts {
+  manifestPath: string;
+  wait: string;
+}
+
+export interface DirectoryOpts {
+  directory: string;
+  wait: string;
+}
+
+export interface ToolingCreateResult {
+  id: string;
+  success: boolean;
+  errors: string[];
+  name: string;
+  message: string;
+}
+
+export interface ToolingRetrieveResult {
```

Comment:
Made some changes to the types to reflect what the api is returning e.g. some values are null when no value is provided